### PR TITLE
Incremental indexing/file delta

### PR DESCRIPTION
**Pull request by AlonsoGuevara**
## Description

Calculate the new inputs for an update run

## Related Issues

[Reference any related issues or tasks that this pull request addresses.]

## Proposed Changes

[List the specific changes made in this pull request.]

## Checklist

- [ ] I have tested these changes locally.
- [ ] I have reviewed the code changes.
- [ ] I have updated the documentation (if necessary).
- [ ] I have added appropriate unit tests (if applicable).

## Additional Notes

[Add any additional notes or context that may be helpful for the reviewer(s).]

### CODE REVIEW

File: graphrag/index/update/files.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+# Output delta named tuple
+class InputDelta(NamedTuple):
+    """Named tuple for output delta."""
+
+    new_inputs: pd.DataFrame
+    deleted_inputs: pd.DataFrame
+
+
+async def get_delta_docs(
```
Comment (andresmor-ms) [inline graphrag/index/update/files.py:22]: does this needs to be async?

Comment (andresmor-ms) [reply graphrag/index/update/files.py:22]: Oh i see you need to use await on _load_table_from_storage

Comment (AlonsoGuevara) [reply graphrag/index/update/files.py:22]: Yup :( All storage related operations are async on the implementations

File: graphrag/index/update/files.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    """
+    final_docs = await _load_table_from_storage(
+        "create_final_documents.parquet", storage
+    )
+
+    # Select distinct title from final docs and from dataset
+    previous_docs: list[str] = final_docs["title"].unique().tolist()
+    dataset_docs: list[str] = input_dataset["title"].unique().tolist()
+
+    # Get the new documents (using loc to ensure DataFrame)
```
Comment (andresmor-ms) [inline graphrag/index/update/files.py:55]: also avoids possible dataframe copies :)

**Review by andresmor-ms — APPROVED** [review]

**Review by Boy124578 — APPROVED** [review]

**Pull request merged**